src/Illuminate/Contracts/Session/Session.php

```diff
@@ -71,6 +71,14 @@ public function exists($key);
      */
     public function has($key);
 
+    /**
+     * Checks if a key is missing or is null.
+     *
+     * @param  string|array  $key
+     * @return bool
+     */
+    public function doesntHave($key);
```

Comment:
I don't think we should add this method to the contract. Seems unnecessary since the opposite one already exists.

Reply:
Agree. `doesntHave` on Eloquent creates a custom query, here it's just a negation of truth.

Reply:
@driesvints thanks, so i'll send this to 9.x instead.